ldpd/lde.c

```diff
@@ -1687,6 +1785,278 @@ lde_change_host_label(int af)
+			if (me && lde_acl_check(acl_name, af, prefix, plen)
+				!= FILTER_PERMIT) {
+				/* This fec is now filtered so send withdraw */
+				lde_send_labelwithdraw(ln, fn, NULL, NULL);
+				lde_map_del(ln, me, 1);
+			}
+			else
+				/* This fec is now allowed in filter, so send
+				 * labels
+				 */
+				if (fn->local_label != NO_LABEL)
+					lde_send_labelmapping(ln, fn, 0);
+		}
+	}
+	RB_FOREACH(ln, nbr_tree, &lde_nbrs)
+		lde_imsg_compose_ldpe(IMSG_MAPPING_ADD_END, ln->peerid, 0,
+			NULL, 0);
+}
+
+void
+lde_change_accept_for_filter(int af)
+{
+	struct lde_nbr  *ln;
+	struct fec      *f;
+	struct fec_node *fn;
+	char            *acl_name;
+	union ldpd_addr *prefix;
+	uint8_t          plen;
+	struct lde_map  *me;
+	struct fec_nh	*fnh;
+	int		 connected = 0;
+
+	/* accept labels from neighbors for fecs defined in this filter */
+	RB_FOREACH(f, fec_tree, &ft) {
+		fn = (struct fec_node *)f;
+		switch (af) {
+		case AF_INET:
+			if (fn->fec.type != FEC_TYPE_IPV4)
+				continue;
+			acl_name = ldeconf->ipv4.acl_label_accept_for;
+			prefix = (union ldpd_addr *)&fn->fec.u.ipv4.prefix;
+			plen = fn->fec.u.ipv4.prefixlen;
+			break;
+		case AF_INET6:
+			if (fn->fec.type != FEC_TYPE_IPV6)
+				continue;
+			acl_name = ldeconf->ipv6.acl_label_accept_for;
+			prefix = (union ldpd_addr *)&fn->fec.u.ipv6.prefix;
+			plen = fn->fec.u.ipv6.prefixlen;
+			break;
+		default:
+			fatalx("lde_change_egress_label: unknown af");
+		}
+
+		connected = 0;
+		if (lde_acl_check(acl_name, af, prefix, plen) != FILTER_PERMIT) {
+			/* fec no longer allowed by this filter remove
+			 * from recv list
+			 */
+			RB_FOREACH(ln, nbr_tree, &lde_nbrs) {
+				me = (struct lde_map *)fec_find(&ln->recv_map,
+					&fn->fec);
+				if (me)
+					lde_map_del(ln, me, 0);
+			}
+		}
+		else {
+			/* fec is now allowed by this filter and if not
+			 * on recv list and not locally connected add it.
+			 */
+			RB_FOREACH(ln, nbr_tree, &lde_nbrs) {
+				me = (struct lde_map *)fec_find(&ln->recv_map,
+					&fn->fec);
+				LIST_FOREACH(fnh, &fn->nexthops, entry)
+					if (fnh->flags & F_FEC_NH_CONNECTED) {
+						connected = 1;
+						break;
+					}
+
+				if (!me && !connected)
+					lde_send_labelrequest(ln, fn, NULL, 1);
+			}
+		}
+	}
+}
+
+void
+lde_change_accept_from_filter(int af)
+{
+	struct lde_nbr  *ln;
+	struct fec      *f;
+	struct fec_node *fn;
+	char            *acl_name;
+	uint8_t          plen;
+	struct lde_map  *me;
+	enum fec_type   type;
+
+	/* accept labels from neighbors specified in this filter */
+	switch (af) {
+	case AF_INET:
+		acl_name = ldeconf->ipv4.acl_label_accept_from;
+		type = FEC_TYPE_IPV4;
+		plen = 32;
+		break;
+	case AF_INET6:
+		acl_name = ldeconf->ipv6.acl_label_accept_from;
+		type = FEC_TYPE_IPV6;
+		plen = 128;
+		break;
+	default:
+		fatalx("lde_change_accept_to_filter: unknown af");
+	}
+	RB_FOREACH(ln, nbr_tree, &lde_nbrs) {
+		if (lde_acl_check(acl_name, af, (union ldpd_addr *)&ln->id, plen)
```

Comment:
This should be `32` (or `IPV4_MAX_BITLEN`) instead of `plen` since `ln->id` is a router-id.

Reply:
fixed

Reply:
i stayed with 32 as it is used in other places in file.    I couldn't find IPV4_MAX_BITLEN used in ldpd

Reply:
Please use a #define and the appropriate define in this case is IPV4_MAX_BITLEN as Renato stated.

Reply:
changed to use define